consensus/hotstuff/signature/randombeacon_inspector.go

```diff
@@ -33,7 +33,7 @@ func NewRandomBeaconInspector(
 		message,
 		signature.RandomBeaconTag)
 	if err != nil {
-		if crypto.IsInvalidInputsError(err) {
+		if crypto.IsInvalidInputsError(err) || crypto.IsNotBLSKeyError(err) {
```

Comment:
As mentioned above, the new crypto version comes with breaking changes. In particular, new crypto specific sentinel errors have been defined, and are replacing the generic `IsInvalidInputsError` (more details this PR description https://github.com/onflow/flow-go/pull/2861). `IsInvalidInputsError` is still used for other error cases for which sentinels are not defined. 

The crypto functions now can return multiple errors depending on the specific invalid behaviour. Not all functions return the same sentinel errors and that's why the generic `IsInvalidInputsError` is replaced with different checks, depending on the function called and the context. The crypto public function docs list the cases and errors returned and should be checked by developers.
As the FVM codeowner, maybe you could validate the error changes in the fvm file.

Reply:
I don't see why this needs to be a breaking change.

InvalidInputError can and should internally call the new predicates since these are also input errors.

anyways.  i've unblock my request for change.

Reply:
There is a longer back-story to this. In a nutshell, we realized that using only `InvalidInputsError` is too coarse and makes it difficult for higher-level code to differentiate internal implementation bugs from byzantine inputs.

I agree that we could still keep the `InvalidInputsError` as an umbrella error, which wraps more specific error cases. But the code interacting with the lower-level crypto implementations is actually not that much. Then the question becomes: it it worth-while to expose an error hierarchy through the API, or is it maybe less work and less complexity to just do a breaking change? It is not really apparent to me, which is less work-intensive. But I feel Tarak's choice to go with the breaking change is certainly a viable approach.

Reply:
From the crypto module perspective, these are all invalid inputs error. From the Flow protocol perspective, these errors can be triggered by different scenarios, and the node has to react differently depending on the scenario detected (as part of the BFT milestone). The sentinel errors break down the original `InvalidInputsError` to help the node identify the issue (malicious behaviour detected/unexpected error..). 
The breaking changes in the crypto module force the protocol code to be updated and to not rely only on the generic inputs error.

Reply:
From the crypto module perspective, these are all invalid inputs error. From the Flow protocol perspective, these errors can be triggered by different scenarios, and the sentinel errors break down the original InvalidInputsError to help the node identify the issue.

Initially, I used an "umbrella error" and sub-error types to avoid the breaking change. That meant the higher layer would need to call multiple embedded error checks to identify the issue. I have finally opted for simpler errors to simplify the checks on the higher layer. Moreover, the breaking change forces integrating the new crypto module version to look at all cases carefully, instead of relying on the "umbrella error" and missing some important sub-cases.